src/test/librbd/mirror/snapshot/test_mock_CreatePrimaryRequest.cc

```diff
@@ -109,8 +109,10 @@ class TestMockMirrorSnapshotCreatePrimaryRequest : public TestMockFixture {
   uint64_t m_snap_seq = 0;
 
   void snap_create(MockTestImageCtx &mock_image_ctx,
-                   const cls::rbd::SnapshotNamespace &ns,
+                   cls::rbd::MirrorSnapshotNamespace &ns,
+                   bool complete,
```

Comment:
The new parameter doesn't make sense to me.  "Completeness" is reflected in the mirror snapshot namespace, which is already passed as `ns`.  If needed, the caller should just set `complete` before invoking `snap_create`.

Reply:
I wanted it to look realistic, i.e set ns.complete after snap_create() call.
Sure, noted this can be optimized. I will fix it as part of the next refresh.